drivers/clock_control/Kconfig

```diff
@@ -9,6 +9,13 @@
 #
 # Clock controller drivers
 #
+
+config CLOCK_CONTROL_MUST_USE_OUT_OF_TREE_DRIVER
```

Comment:
hmmm, why do we need to have Kconfigs for this? Are we going to have such a kconfig for each driver subsystem? Are we talking about 2 conflicting drivers and we would select the one out of tree instead  of the one in the tree? In short, this looks like a hack to me. Need some documentation or a design proposal explaining how this can be done in a generic way without polluting Kconfig with such options.

Reply:
> Need some documentation or a design proposal explaining how this can be done in a generic way without polluting Kconfig with such options.

Please prove me wrong @ulfalizer, it is unavoidable to pollute the kconfig with these options without resolving #8181, which is not going to happen any time soon.

For your other questions, please see https://github.com/zephyrproject-rtos/zephyr/pull/13631/commits/8f1e6cb177e2f8fc49b0830589439d92c4ecf260

Reply:
The reasoning is described in this commit message: https://github.com/zephyrproject-rtos/zephyr/pull/13631/commits/8f1e6cb177e2f8fc49b0830589439d92c4ecf260

Reply:
Yes we need this, unfortunately. This is for the case were an out-of-tree component for some reason _must_ use it's own driver. There is no way in Kconfig that I and others have found to effectively de-select a Kconfig choice/config without severely polluting the Zephyr configuration files.

> Are we going to have such a kconfig for each driver subsystem?

I would add it at a per use-case/need basis. But this is a generic pattern, yes.

> Are we talking about 2 conflicting drivers and we would select the one out of tree instead of the one in the tree? 

Yes. And the fact that they are not only conflicting but another subsystem _must_ use the out-of-tree one. We also want this switch to happen automatically, so if I select a different BLE controller in the menuconfig, the correct driver dependencies are resolved without the need for other changes by the user.

> In short, this looks like a hack to me.

I would like it for there to be a nicer way of doing this. But as far as I can tell, this is the most pragmatic, least intrusive and nicest way of handling out-of-tree dependencies like this. IMO the only way of actually solving the root problem is to resolve https://github.com/zephyrproject-rtos/zephyr/issues/8181.

> Need some documentation or a design proposal [...]

What would such a proposal look like?  Could I add more to https://github.com/zephyrproject-rtos/zephyr/pull/13631/commits/8f1e6cb177e2f8fc49b0830589439d92c4ecf260 to explain the why and how better?